manifests/package/redhat.pp

```diff
@@ -94,6 +95,14 @@
     }
   }
 
+  if $dnfmodule and ($facts['os']['name'] in ['RedHat', 'CentOS', 'VirtuozzoLinux']) and ($facts['os']['release']['major'] in ['8']) {
```

Comment:
Prefer `fact('os.name')` ([from stdlib](https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#fact)) or `$facts.get('os.name')` ([from Puppet core](https://puppet.com/docs/puppet/7/function.html#get)) over `$facts['os']['name']`.

Also, regarding version comparison, unless you are sure version 9, 10 and so on will not be concerned, better rely on [`versioncmp`](https://puppet.com/docs/puppet/7/function.html#versioncmp), e.g. `versioncmp(fact('os.release.full'), '8.0') >= 0`

Reply:
I changed to fact('os.name') and as EL 9, 10 etc. is not released I took your suggestion with versioncmp.